apps/sugaming-discord/src/cs2-teams/cs2-teams.controller.ts

```diff
@@ -0,0 +1,73 @@
+import { Controller } from '@nestjs/common';
+import { Cs2TeamsService } from '@sugaming/sugaming-services/cs2/teams/cs2-teams.service';
+import { InjectDiscordClient } from '@discord-nestjs/core';
+import { Client, ChannelType, Guild } from 'discord.js';
+import { MessagePattern, Payload } from '@nestjs/microservices';
+import { Cs2Team } from '@prisma/client';
+
+@Controller()
+export class Cs2TeamsController {
+  constructor(
+    @InjectDiscordClient() private readonly discordClient: Client,
+    private readonly cs2TeamsService: Cs2TeamsService,
+  ) {}
+
+  @MessagePattern('cs2-teams:team-created')
+  async handleTeamCreatedEvent(@Payload() createdTeam: Cs2Team) {
+    // Get the first guild
+    const guild = await this.discordClient.guilds.cache.first();
+
+    // Create the team category and channels
+    const category = await this.createTeamChannels(createdTeam.name, guild);
+
+    // Create a role for the team
+    const newRole = await guild.roles.create({
```

Comment:
Please refactor this by creating the role before the team channels. Move the role creation logic in the `Cs2TeamsService` in `sugaming-services`. Then, after creating the role, pass it to the team creation function (which should also be located in the `sugaming-services` library).